collect_app/src/main/java/org/odk/collect/android/receivers/NetworkReceiver.java

```diff
@@ -272,6 +255,27 @@ private String formatOverallResultMessage(HashMap<String, String> resultMessages
         return message;
     }
 
+    private void showUploadStatusNotification(HashMap<String, String> resultMessagesByInstanceId, String message) {
+        Intent notifyIntent = new Intent(Collect.getInstance(), NotificationActivity.class);
+        notifyIntent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+        notifyIntent.putExtra(NotificationActivity.NOTIFICATION_TITLE, Collect.getInstance().getString(R.string.upload_results));
+        notifyIntent.putExtra(NotificationActivity.NOTIFICATION_MESSAGE, message.trim());
+
+        PendingIntent pendingNotify = PendingIntent.getActivity(Collect.getInstance(), FORMS_UPLOADED_NOTIFICATION,
+                notifyIntent, PendingIntent.FLAG_UPDATE_CURRENT);
+
+        NotificationCompat.Builder builder = new NotificationCompat.Builder(Collect.getInstance())
+                .setSmallIcon(IconUtils.getNotificationAppIcon())
+                .setContentTitle(Collect.getInstance().getString(R.string.odk_auto_note))
+                .setContentIntent(pendingNotify)
+                .setContentText(getContentText(resultMessagesByInstanceId))
+                .setAutoCancel(true);
+
+        NotificationManager notificationManager = (NotificationManager) Collect.getInstance()
+                .getSystemService(Context.NOTIFICATION_SERVICE);
+        notificationManager.notify(1328974928, builder.build());
```

Comment:
1328974928?  What's that?

Reply:
Clearly, there's more cleanup that can be done. It's an ID unique to this notification (and should be a named value somewhere).